src/main/java/com/karankumar/bookproject/backend/entity/User.java

```diff
@@ -62,6 +62,12 @@
 
     // Note: this is allowed to be null if a user signs up without an email
     @NotEmpty
+    // For the RegExp see https://owasp.org/www-community/OWASP_Validation_Regex_Repository
+    @Pattern(
+            regexp = "^[a-zA-Z0-9_+&*-]+(?:\\.[a-zA-Z0-9_+&*-]+)*@(?:[a-zA-Z0-9-]+\\.)+[a-zA-Z]{2,7}$",
```

Comment:
You seem to be using the wrong number of backslashes at certain points. Could you replace the regex with the following (taken from OWASP):
`[a-zA-Z0-9_+&*-]+(?:\.[a-zA-Z0-9_+&*-]+)*@(?:[a-zA-Z0-9-]+\.)+[a-zA-Z]{2,7}$]]`

Reply:
Thanks for your comments! One backslash is needed to escape the dot and the other is needed to backslash the first. Take a look here: https://stackoverflow.com/a/3675027/7885311.

Reply:
My bad. Ignore me :)